lib/streamlit/runtime/scriptrunner/script_runner.py

```diff
@@ -475,30 +507,35 @@ def _run_script(self, rerun_data: RerunData) -> None:
                 event=ScriptRunnerEvent.SCRIPT_STARTED,
                 page_script_hash=page_script_hash,
                 fragment_ids_this_run=fragment_ids_this_run,
-                pages=self._pages_manager.get_pages(),
+                pages={},
             )
+            if ctx.pages_manager.mpa_version == 1:
```

Comment:
Why is this only needed for v1?

Reply:
Will add a comment but V2 assumes the script will send its own navigation message. V1 does not.